core/app/models/spree/taxon/active_storage_attachment.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require 'active_storage'
+
+module Spree::Taxon::ActiveStorageAttachment
+  extend ActiveSupport::Concern
+
+  module IOAttachmentSupport
+    extend ActiveSupport::Concern
+
+    def icon=(attachable)
```

Comment:
maybe adding more context about why this setter is needed in the commit message could help inexperienced users (like myself 😄) understanding.

Reply:
It's later extracted into `def redefine_attachment_writer_with_legacy_io_support`, if that's not enogh I can add commets to both the original and the extracted version, let me know 👍